pytensor/link/jax/dispatch/slinalg.py

```diff
@@ -54,7 +54,21 @@ def solve(a, b):
             dl = jax.numpy.diagonal(a, offset=-1, axis1=-2, axis2=-1)
             d = jax.numpy.diagonal(a, offset=0, axis1=-2, axis2=-1)
             du = jax.numpy.diagonal(a, offset=1, axis1=-2, axis2=-1)
-            return jax.lax.linalg.tridiagonal_solve(dl, d, du, b, lower=lower)
+            # jax requires dl and du to have the same shape as d
+            dl = jax.numpy.pad(dl, (1, 0))
+            du = jax.numpy.pad(du, (0, 1))
+            # if b is a vector, broadcast it to be a matrix
+            b_is_vec = len(b.shape) == 1
+            if b_is_vec:
+                b = jax.numpy.expand_dims(b, -1)
+
+            res = jax.lax.linalg.tridiagonal_solve(dl, d, du, b)
+
+            if b_is_vec:
+                # if b is a vector, return a vector
+                return res.flatten()
+            else:
```

Comment:
You don't need an else here since the `if` branch returns -- you can just `return res` directly

Reply:
It's clear that it's not needed, but I don't like if code is too implicit -- here it makes little difference, but in general I try to be as explicit about logic as possible because I find it easier to parse when revisiting codebase and it also avoids some gotchas in cases that are less simple than this one.

If that's the general guideline in pytensor to be as short as possible, I'll change it.

Reply:
It's a nit, so you can just ignore. Otherwise a pattern I like is to have a single return:

```python
res = ...
if cond:
  res = f(res)
return res
```